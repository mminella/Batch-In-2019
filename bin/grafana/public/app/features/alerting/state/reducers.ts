import moment from 'moment';
import { AlertRuleDTO, AlertRule, AlertRulesState } from 'app/types';
import { Action, ActionTypes } from './actions';
import alertDef from './alertDef';

export const initialState: AlertRulesState = { items: [], searchQuery: '', isLoading: false };

function convertToAlertRule(rule, state): AlertRule {
  const stateModel = alertDef.getStateDisplayModel(state);
  rule.stateText = stateModel.text;
  rule.stateIcon = stateModel.iconClass;
  rule.stateClass = stateModel.stateClass;
  rule.stateAge = moment(rule.newStateDate)
    .fromNow()
    .replace(' ago', '');

  if (rule.state !== 'paused') {
    if (rule.executionError) {
      rule.info = 'Execution Error: ' + rule.executionError;
    }
    if (rule.evalData && rule.evalData.noData) {
      rule.info = 'Query returned no data';
    }
  }

  return rule;
}

export const alertRulesReducer = (state = initialState, action: Action): AlertRulesState => {
  switch (action.type) {
    case ActionTypes.LoadAlertRules: {
      return { ...state, isLoading: true };
    }

    case ActionTypes.LoadedAlertRules: {
      const alertRules: AlertRuleDTO[] = action.payload;

      const alertRulesViewModel: AlertRule[] = alertRules.map(rule => {
        return convertToAlertRule(rule, rule.state);
      });

      return { ...state, items: alertRulesViewModel, isLoading: false };
    }

    case ActionTypes.SetSearchQuery:
      return { ...state, searchQuery: action.payload };
  }

  return state;
};

export default {
  alertRules: alertRulesReducer,
};
