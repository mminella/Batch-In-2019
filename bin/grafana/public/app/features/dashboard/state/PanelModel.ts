// Libraries
import _ from 'lodash';

// Types
import { Emitter } from 'app/core/utils/emitter';
import { DataQuery, TimeSeries, Threshold } from '@grafana/ui';
import { TableData } from '@grafana/ui/src';

export interface GridPos {
  x: number;
  y: number;
  w: number;
  h: number;
  static?: boolean;
}

const notPersistedProperties: { [str: string]: boolean } = {
  events: true,
  fullscreen: true,
  isEditing: true,
  hasRefreshed: true,
  cachedPluginOptions: true,
};

// For angular panels we need to clean up properties when changing type
// To make sure the change happens without strange bugs happening when panels use same
// named property with different type / value expectations
// This is not required for react panels

const mustKeepProps: { [str: string]: boolean } = {
  id: true,
  gridPos: true,
  type: true,
  title: true,
  scopedVars: true,
  repeat: true,
  repeatIteration: true,
  repeatPanelId: true,
  repeatDirection: true,
  repeatedByRow: true,
  minSpan: true,
  collapsed: true,
  panels: true,
  targets: true,
  datasource: true,
  timeFrom: true,
  timeShift: true,
  hideTimeOverride: true,
  description: true,
  links: true,
  fullscreen: true,
  isEditing: true,
  hasRefreshed: true,
  events: true,
  cacheTimeout: true,
  cachedPluginOptions: true,
  transparent: true,
};

const defaults: any = {
  gridPos: { x: 0, y: 0, h: 3, w: 6 },
  datasource: null,
  targets: [{ refId: 'A' }],
  cachedPluginOptions: {},
  transparent: false,
};

export class PanelModel {
  id: number;
  gridPos: GridPos;
  type: string;
  title: string;
  alert?: any;
  scopedVars?: any;
  repeat?: string;
  repeatIteration?: number;
  repeatPanelId?: number;
  repeatDirection?: string;
  repeatedByRow?: boolean;
  maxPerRow?: number;
  collapsed?: boolean;
  panels?: any;
  soloMode?: boolean;
  targets: DataQuery[];
  datasource: string;
  thresholds?: any;

  snapshotData?: TimeSeries[] | [TableData];
  timeFrom?: any;
  timeShift?: any;
  hideTimeOverride?: any;
  options: {
    [key: string]: any;
  };

  maxDataPoints?: number;
  interval?: string;
  description?: string;
  links?: [];
  transparent: boolean;

  // non persisted
  fullscreen: boolean;
  isEditing: boolean;
  hasRefreshed: boolean;
  events: Emitter;
  cacheTimeout?: any;
  cachedPluginOptions?: any;

  constructor(model) {
    this.events = new Emitter();

    // copy properties from persisted model
    for (const property in model) {
      this[property] = model[property];
    }

    // defaults
    _.defaultsDeep(this, _.cloneDeep(defaults));
    // queries must have refId
    this.ensureQueryIds();

    this.restoreInfintyForThresholds();
  }

  ensureQueryIds() {
    if (this.targets) {
      for (const query of this.targets) {
        if (!query.refId) {
          query.refId = this.getNextQueryLetter();
        }
      }
    }
  }

  restoreInfintyForThresholds() {
    if (this.options && this.options.thresholds) {
      this.options.thresholds = this.options.thresholds.map((threshold: Threshold) => {
        // JSON serialization of -Infinity is 'null' so lets convert it back to -Infinity
        if (threshold.index === 0 && threshold.value === null) {
          return { ...threshold, value: -Infinity };
        }

        return threshold;
      });
    }
  }

  getOptions(panelDefaults) {
    return _.defaultsDeep(this.options || {}, panelDefaults);
  }

  updateOptions(options: object) {
    this.options = options;
    this.render();
  }

  getSaveModel() {
    const model: any = {};
    for (const property in this) {
      if (notPersistedProperties[property] || !this.hasOwnProperty(property)) {
        continue;
      }

      if (_.isEqual(this[property], defaults[property])) {
        continue;
      }

      model[property] = _.cloneDeep(this[property]);
    }

    return model;
  }

  setViewMode(fullscreen: boolean, isEditing: boolean) {
    this.fullscreen = fullscreen;
    this.isEditing = isEditing;
    this.events.emit('view-mode-changed');
  }

  updateGridPos(newPos: GridPos) {
    let sizeChanged = false;

    if (this.gridPos.w !== newPos.w || this.gridPos.h !== newPos.h) {
      sizeChanged = true;
    }

    this.gridPos.x = newPos.x;
    this.gridPos.y = newPos.y;
    this.gridPos.w = newPos.w;
    this.gridPos.h = newPos.h;

    if (sizeChanged) {
      this.events.emit('panel-size-changed');
    }
  }

  resizeDone() {
    this.events.emit('panel-size-changed');
  }

  refresh() {
    this.hasRefreshed = true;
    this.events.emit('refresh');
  }

  render() {
    if (!this.hasRefreshed) {
      this.refresh();
    } else {
      this.events.emit('render');
    }
  }

  initialized() {
    this.events.emit('panel-initialized');
  }

  private getOptionsToRemember() {
    return Object.keys(this).reduce((acc, property) => {
      if (notPersistedProperties[property] || mustKeepProps[property]) {
        return acc;
      }
      return {
        ...acc,
        [property]: this[property],
      };
    }, {});
  }

  private saveCurrentPanelOptions() {
    this.cachedPluginOptions[this.type] = this.getOptionsToRemember();
  }

  private restorePanelOptions(pluginId: string) {
    const prevOptions = this.cachedPluginOptions[pluginId] || {};

    Object.keys(prevOptions).map(property => {
      this[property] = prevOptions[property];
    });
  }

  changeType(pluginId: string, fromAngularPanel: boolean) {
    this.saveCurrentPanelOptions();
    this.type = pluginId;

    // for angular panels only we need to remove all events and let angular panels do some cleanup
    if (fromAngularPanel) {
      this.destroy();
    }

    // remove panel type specific  options
    for (const key of _.keys(this)) {
      if (mustKeepProps[key]) {
        continue;
      }

      delete this[key];
    }

    this.restorePanelOptions(pluginId);
  }

  addQuery(query?: Partial<DataQuery>) {
    query = query || { refId: 'A' };
    query.refId = this.getNextQueryLetter();
    this.targets.push(query as DataQuery);
  }

  getNextQueryLetter(): string {
    const letters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ';

    return _.find(letters, refId => {
      return _.every(this.targets, other => {
        return other.refId !== refId;
      });
    });
  }

  changeQuery(query: DataQuery, index: number) {
    // ensure refId is maintained
    query.refId = this.targets[index].refId;

    // update query in array
    this.targets = this.targets.map((item, itemIndex) => {
      if (itemIndex === index) {
        return query;
      }
      return item;
    });
  }

  destroy() {
    this.events.emit('panel-teardown');
    this.events.removeAllListeners();
  }
}
