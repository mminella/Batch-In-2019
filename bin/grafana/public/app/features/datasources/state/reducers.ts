import { DataSourcesState, Plugin } from 'app/types';
import { DataSourceSettings } from '@grafana/ui/src/types';
import {
  dataSourceLoaded,
  dataSourcesLoaded,
  setDataSourcesSearchQuery,
  setDataSourcesLayoutMode,
  dataSourceTypesLoad,
  dataSourceTypesLoaded,
  setDataSourceTypeSearchQuery,
  dataSourceMetaLoaded,
  setDataSourceName,
  setIsDefault,
} from './actions';
import { LayoutModes } from 'app/core/components/LayoutSelector/LayoutSelector';
import { reducerFactory } from 'app/core/redux';

export const initialState: DataSourcesState = {
  dataSources: [],
  dataSource: {} as DataSourceSettings,
  layoutMode: LayoutModes.List,
  searchQuery: '',
  dataSourcesCount: 0,
  dataSourceTypes: [],
  dataSourceTypeSearchQuery: '',
  hasFetched: false,
  isLoadingDataSources: false,
  dataSourceMeta: {} as Plugin,
};

export const dataSourcesReducer = reducerFactory(initialState)
  .addMapper({
    filter: dataSourcesLoaded,
    mapper: (state, action) => ({
      ...state,
      hasFetched: true,
      dataSources: action.payload,
      dataSourcesCount: action.payload.length,
    }),
  })
  .addMapper({
    filter: dataSourceLoaded,
    mapper: (state, action) => ({ ...state, dataSource: action.payload }),
  })
  .addMapper({
    filter: setDataSourcesSearchQuery,
    mapper: (state, action) => ({ ...state, searchQuery: action.payload }),
  })
  .addMapper({
    filter: setDataSourcesLayoutMode,
    mapper: (state, action) => ({ ...state, layoutMode: action.payload }),
  })
  .addMapper({
    filter: dataSourceTypesLoad,
    mapper: state => ({ ...state, dataSourceTypes: [], isLoadingDataSources: true }),
  })
  .addMapper({
    filter: dataSourceTypesLoaded,
    mapper: (state, action) => ({
      ...state,
      dataSourceTypes: action.payload,
      isLoadingDataSources: false,
    }),
  })
  .addMapper({
    filter: setDataSourceTypeSearchQuery,
    mapper: (state, action) => ({ ...state, dataSourceTypeSearchQuery: action.payload }),
  })
  .addMapper({
    filter: dataSourceMetaLoaded,
    mapper: (state, action) => ({ ...state, dataSourceMeta: action.payload }),
  })
  .addMapper({
    filter: setDataSourceName,
    mapper: (state, action) => ({ ...state, dataSource: { ...state.dataSource, name: action.payload } }),
  })
  .addMapper({
    filter: setIsDefault,
    mapper: (state, action) => ({
      ...state,
      dataSource: { ...state.dataSource, isDefault: action.payload },
    }),
  })
  .create();

export default {
  dataSources: dataSourcesReducer,
};
