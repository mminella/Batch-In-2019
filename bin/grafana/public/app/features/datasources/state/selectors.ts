import { DataSourceSettings } from '@grafana/ui/src/types';

export const getDataSources = state => {
  const regex = new RegExp(state.searchQuery, 'i');

  return state.dataSources.filter(dataSource => {
    return regex.test(dataSource.name) || regex.test(dataSource.database);
  });
};

export const getDataSourceTypes = state => {
  const regex = new RegExp(state.dataSourceTypeSearchQuery, 'i');

  return state.dataSourceTypes.filter(type => {
    return regex.test(type.name);
  });
};

export const getDataSource = (state, dataSourceId): DataSourceSettings | null => {
  if (state.dataSource.id === parseInt(dataSourceId, 10)) {
    return state.dataSource;
  }
  return {} as DataSourceSettings;
};

export const getDataSourceMeta = (state, type): Plugin => {
  if (state.dataSourceMeta.id === type) {
    return state.dataSourceMeta;
  }

  return {} as Plugin;
};

export const getDataSourcesSearchQuery = state => state.searchQuery;
export const getDataSourcesLayoutMode = state => state.layoutMode;
export const getDataSourcesCount = state => state.dataSourcesCount;
