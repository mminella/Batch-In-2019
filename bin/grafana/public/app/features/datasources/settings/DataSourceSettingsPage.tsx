// Libraries
import React, { PureComponent } from 'react';
import { hot } from 'react-hot-loader';
import { connect } from 'react-redux';

// Components
import Page from 'app/core/components/Page/Page';
import PluginSettings from './PluginSettings';
import BasicSettings from './BasicSettings';
import ButtonRow from './ButtonRow';

// Services & Utils
import appEvents from 'app/core/app_events';
import { getBackendSrv } from 'app/core/services/backend_srv';
import { getDatasourceSrv } from 'app/features/plugins/datasource_srv';

// Actions & selectors
import { getDataSource, getDataSourceMeta } from '../state/selectors';
import { deleteDataSource, loadDataSource, setDataSourceName, setIsDefault, updateDataSource } from '../state/actions';
import { getNavModel } from 'app/core/selectors/navModel';
import { getRouteParamsId } from 'app/core/selectors/location';

// Types
import { NavModel, Plugin, StoreState } from 'app/types/';
import { DataSourceSettings } from '@grafana/ui/src/types/';
import { getDataSourceLoadingNav } from '../state/navModel';

export interface Props {
  navModel: NavModel;
  dataSource: DataSourceSettings;
  dataSourceMeta: Plugin;
  pageId: number;
  deleteDataSource: typeof deleteDataSource;
  loadDataSource: typeof loadDataSource;
  setDataSourceName: typeof setDataSourceName;
  updateDataSource: typeof updateDataSource;
  setIsDefault: typeof setIsDefault;
}

interface State {
  dataSource: DataSourceSettings;
  isTesting?: boolean;
  testingMessage?: string;
  testingStatus?: string;
}

enum DataSourceStates {
  Alpha = 'alpha',
  Beta = 'beta',
}

export class DataSourceSettingsPage extends PureComponent<Props, State> {
  constructor(props: Props) {
    super(props);

    this.state = {
      dataSource: {} as DataSourceSettings,
    };
  }

  async componentDidMount() {
    const { loadDataSource, pageId } = this.props;

    await loadDataSource(pageId);
  }

  componentDidUpdate(prevProps: Props) {
    const { dataSource } = this.props;

    if (prevProps.dataSource !== dataSource) {
      this.setState({ dataSource });
    }
  }

  onSubmit = async (evt: React.FormEvent<HTMLFormElement>) => {
    evt.preventDefault();

    await this.props.updateDataSource({ ...this.state.dataSource, name: this.props.dataSource.name });

    this.testDataSource();
  };

  onTest = async (evt: React.FormEvent<HTMLFormElement>) => {
    evt.preventDefault();

    this.testDataSource();
  };

  onDelete = () => {
    appEvents.emit('confirm-modal', {
      title: 'Delete',
      text: 'Are you sure you want to delete this data source?',
      yesText: 'Delete',
      icon: 'fa-trash',
      onConfirm: () => {
        this.confirmDelete();
      },
    });
  };

  confirmDelete = () => {
    this.props.deleteDataSource();
  };

  onModelChange = (dataSource: DataSourceSettings) => {
    this.setState({ dataSource });
  };

  isReadOnly() {
    return this.props.dataSource.readOnly === true;
  }

  shouldRenderInfoBox() {
    const { state } = this.props.dataSourceMeta;

    return state === DataSourceStates.Alpha || state === DataSourceStates.Beta;
  }

  getInfoText() {
    const { dataSourceMeta } = this.props;

    switch (dataSourceMeta.state) {
      case DataSourceStates.Alpha:
        return (
          'This plugin is marked as being in alpha state, which means it is in early development phase and updates' +
          ' will include breaking changes.'
        );

      case DataSourceStates.Beta:
        return (
          'This plugin is marked as being in a beta development state. This means it is in currently in active' +
          ' development and could be missing important features.'
        );
    }

    return null;
  }

  renderIsReadOnlyMessage() {
    return (
      <div className="grafana-info-box span8">
        This datasource was added by config and cannot be modified using the UI. Please contact your server admin to
        update this datasource.
      </div>
    );
  }

  async testDataSource() {
    const dsApi = await getDatasourceSrv().get(this.state.dataSource.name);

    if (!dsApi.testDatasource) {
      return;
    }

    this.setState({ isTesting: true, testingMessage: 'Testing...', testingStatus: 'info' });

    getBackendSrv().withNoBackendCache(async () => {
      try {
        const result = await dsApi.testDatasource();

        this.setState({
          isTesting: false,
          testingStatus: result.status,
          testingMessage: result.message,
        });
      } catch (err) {
        let message = '';

        if (err.statusText) {
          message = 'HTTP Error ' + err.statusText;
        } else {
          message = err.message;
        }

        this.setState({
          isTesting: false,
          testingStatus: 'error',
          testingMessage: message,
        });
      }
    });
  }

  get hasDataSource() {
    return Object.keys(this.props.dataSource).length > 0;
  }

  render() {
    const { dataSource, dataSourceMeta, navModel, setDataSourceName, setIsDefault } = this.props;
    const { testingMessage, testingStatus } = this.state;

    return (
      <Page navModel={navModel}>
        <Page.Contents isLoading={!this.hasDataSource}>
          {this.hasDataSource && (
            <div>
              <form onSubmit={this.onSubmit}>
                {this.isReadOnly() && this.renderIsReadOnlyMessage()}
                {this.shouldRenderInfoBox() && <div className="grafana-info-box">{this.getInfoText()}</div>}

                <BasicSettings
                  dataSourceName={dataSource.name}
                  isDefault={dataSource.isDefault}
                  onDefaultChange={state => setIsDefault(state)}
                  onNameChange={name => setDataSourceName(name)}
                />

                {dataSourceMeta.module && (
                  <PluginSettings
                    dataSource={dataSource}
                    dataSourceMeta={dataSourceMeta}
                    onModelChange={this.onModelChange}
                  />
                )}

                <div className="gf-form-group">
                  {testingMessage && (
                    <div className={`alert-${testingStatus} alert`}>
                      <div className="alert-icon">
                        {testingStatus === 'error' ? (
                          <i className="fa fa-exclamation-triangle" />
                        ) : (
                          <i className="fa fa-check" />
                        )}
                      </div>
                      <div className="alert-body">
                        <div className="alert-title">{testingMessage}</div>
                      </div>
                    </div>
                  )}
                </div>

                <ButtonRow
                  onSubmit={event => this.onSubmit(event)}
                  isReadOnly={this.isReadOnly()}
                  onDelete={this.onDelete}
                  onTest={event => this.onTest(event)}
                />
              </form>
            </div>
          )}
        </Page.Contents>
      </Page>
    );
  }
}

function mapStateToProps(state: StoreState) {
  const pageId = getRouteParamsId(state.location);
  const dataSource = getDataSource(state.dataSources, pageId);

  return {
    navModel: getNavModel(state.navIndex, `datasource-settings-${pageId}`, getDataSourceLoadingNav('settings')),
    dataSource: getDataSource(state.dataSources, pageId),
    dataSourceMeta: getDataSourceMeta(state.dataSources, dataSource.type),
    pageId: pageId,
  };
}

const mapDispatchToProps = {
  deleteDataSource,
  loadDataSource,
  setDataSourceName,
  updateDataSource,
  setIsDefault,
};

export default hot(module)(
  connect(
    mapStateToProps,
    mapDispatchToProps
  )(DataSourceSettingsPage)
);
