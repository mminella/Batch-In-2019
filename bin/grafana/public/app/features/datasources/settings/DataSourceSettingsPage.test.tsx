import React from 'react';
import { shallow } from 'enzyme';
import { DataSourceSettingsPage, Props } from './DataSourceSettingsPage';
import { NavModel } from 'app/types';
import { DataSourceSettings } from '@grafana/ui';
import { getMockDataSource } from '../__mocks__/dataSourcesMocks';
import { getMockPlugin } from '../../plugins/__mocks__/pluginMocks';
import { setDataSourceName, setIsDefault } from '../state/actions';

const setup = (propOverrides?: object) => {
  const props: Props = {
    navModel: {} as NavModel,
    dataSource: getMockDataSource(),
    dataSourceMeta: getMockPlugin(),
    pageId: 1,
    deleteDataSource: jest.fn(),
    loadDataSource: jest.fn(),
    setDataSourceName,
    updateDataSource: jest.fn(),
    setIsDefault,
  };

  Object.assign(props, propOverrides);

  return shallow(<DataSourceSettingsPage {...props} />);
};

describe('Render', () => {
  it('should render component', () => {
    const wrapper = setup();

    expect(wrapper).toMatchSnapshot();
  });

  it('should render loader', () => {
    const wrapper = setup({
      dataSource: {} as DataSourceSettings,
    });

    expect(wrapper).toMatchSnapshot();
  });

  it('should render beta info text', () => {
    const wrapper = setup({
      dataSourceMeta: { ...getMockPlugin(), state: 'beta' },
    });

    expect(wrapper).toMatchSnapshot();
  });

  it('should render alpha info text', () => {
    const wrapper = setup({
      dataSourceMeta: { ...getMockPlugin(), state: 'alpha' },
    });

    expect(wrapper).toMatchSnapshot();
  });

  it('should render is ready only message', () => {
    const wrapper = setup({
      dataSource: { ...getMockDataSource(), readOnly: true },
    });

    expect(wrapper).toMatchSnapshot();
  });
});
