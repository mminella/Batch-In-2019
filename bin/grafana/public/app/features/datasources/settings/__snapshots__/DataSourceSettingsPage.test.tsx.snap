// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render should render alpha info text 1`] = `
<Page
  navModel={Object {}}
>
  <PageContents
    isLoading={false}
  >
    <div>
      <form
        onSubmit={[Function]}
      >
        <div
          className="grafana-info-box"
        >
          This plugin is marked as being in alpha state, which means it is in early development phase and updates will include breaking changes.
        </div>
        <BasicSettings
          dataSourceName="gdev-cloudwatch"
          isDefault={false}
          onDefaultChange={[Function]}
          onNameChange={[Function]}
        />
        <PluginSettings
          dataSource={
            Object {
              "access": "",
              "basicAuth": false,
              "basicAuthPassword": "",
              "basicAuthUser": "",
              "database": "",
              "id": 13,
              "isDefault": false,
              "jsonData": Object {
                "authType": "credentials",
                "defaultRegion": "eu-west-2",
              },
              "name": "gdev-cloudwatch",
              "orgId": 1,
              "password": "",
              "readOnly": false,
              "type": "cloudwatch",
              "typeLogoUrl": "public/app/plugins/datasource/cloudwatch/img/amazon-web-services.png",
              "url": "",
              "user": "",
              "withCredentials": false,
            }
          }
          dataSourceMeta={
            Object {
              "defaultNavUrl": "some/url",
              "enabled": false,
              "hasUpdate": false,
              "id": "1",
              "info": Object {
                "author": Object {
                  "name": "Grafana Labs",
                  "url": "url/to/GrafanaLabs",
                },
                "description": "pretty decent plugin",
                "links": Array [
                  Object {
                    "name": "project",
                    "url": "one link",
                  },
                ],
                "logos": Object {
                  "large": "large/logo",
                  "small": "small/logo",
                },
                "screenshots": Array [
                  Object {
                    "path": "screenshot",
                  },
                ],
                "updated": "2018-09-26",
                "version": "1",
              },
              "latestVersion": "1",
              "module": Object {},
              "name": "pretty cool plugin 1",
              "pinned": false,
              "state": "alpha",
              "type": "",
            }
          }
          onModelChange={[Function]}
        />
        <div
          className="gf-form-group"
        />
        <ButtonRow
          isReadOnly={false}
          onDelete={[Function]}
          onSubmit={[Function]}
          onTest={[Function]}
        />
      </form>
    </div>
  </PageContents>
</Page>
`;

exports[`Render should render beta info text 1`] = `
<Page
  navModel={Object {}}
>
  <PageContents
    isLoading={false}
  >
    <div>
      <form
        onSubmit={[Function]}
      >
        <div
          className="grafana-info-box"
        >
          This plugin is marked as being in a beta development state. This means it is in currently in active development and could be missing important features.
        </div>
        <BasicSettings
          dataSourceName="gdev-cloudwatch"
          isDefault={false}
          onDefaultChange={[Function]}
          onNameChange={[Function]}
        />
        <PluginSettings
          dataSource={
            Object {
              "access": "",
              "basicAuth": false,
              "basicAuthPassword": "",
              "basicAuthUser": "",
              "database": "",
              "id": 13,
              "isDefault": false,
              "jsonData": Object {
                "authType": "credentials",
                "defaultRegion": "eu-west-2",
              },
              "name": "gdev-cloudwatch",
              "orgId": 1,
              "password": "",
              "readOnly": false,
              "type": "cloudwatch",
              "typeLogoUrl": "public/app/plugins/datasource/cloudwatch/img/amazon-web-services.png",
              "url": "",
              "user": "",
              "withCredentials": false,
            }
          }
          dataSourceMeta={
            Object {
              "defaultNavUrl": "some/url",
              "enabled": false,
              "hasUpdate": false,
              "id": "1",
              "info": Object {
                "author": Object {
                  "name": "Grafana Labs",
                  "url": "url/to/GrafanaLabs",
                },
                "description": "pretty decent plugin",
                "links": Array [
                  Object {
                    "name": "project",
                    "url": "one link",
                  },
                ],
                "logos": Object {
                  "large": "large/logo",
                  "small": "small/logo",
                },
                "screenshots": Array [
                  Object {
                    "path": "screenshot",
                  },
                ],
                "updated": "2018-09-26",
                "version": "1",
              },
              "latestVersion": "1",
              "module": Object {},
              "name": "pretty cool plugin 1",
              "pinned": false,
              "state": "beta",
              "type": "",
            }
          }
          onModelChange={[Function]}
        />
        <div
          className="gf-form-group"
        />
        <ButtonRow
          isReadOnly={false}
          onDelete={[Function]}
          onSubmit={[Function]}
          onTest={[Function]}
        />
      </form>
    </div>
  </PageContents>
</Page>
`;

exports[`Render should render component 1`] = `
<Page
  navModel={Object {}}
>
  <PageContents
    isLoading={false}
  >
    <div>
      <form
        onSubmit={[Function]}
      >
        <BasicSettings
          dataSourceName="gdev-cloudwatch"
          isDefault={false}
          onDefaultChange={[Function]}
          onNameChange={[Function]}
        />
        <PluginSettings
          dataSource={
            Object {
              "access": "",
              "basicAuth": false,
              "basicAuthPassword": "",
              "basicAuthUser": "",
              "database": "",
              "id": 13,
              "isDefault": false,
              "jsonData": Object {
                "authType": "credentials",
                "defaultRegion": "eu-west-2",
              },
              "name": "gdev-cloudwatch",
              "orgId": 1,
              "password": "",
              "readOnly": false,
              "type": "cloudwatch",
              "typeLogoUrl": "public/app/plugins/datasource/cloudwatch/img/amazon-web-services.png",
              "url": "",
              "user": "",
              "withCredentials": false,
            }
          }
          dataSourceMeta={
            Object {
              "defaultNavUrl": "some/url",
              "enabled": false,
              "hasUpdate": false,
              "id": "1",
              "info": Object {
                "author": Object {
                  "name": "Grafana Labs",
                  "url": "url/to/GrafanaLabs",
                },
                "description": "pretty decent plugin",
                "links": Array [
                  Object {
                    "name": "project",
                    "url": "one link",
                  },
                ],
                "logos": Object {
                  "large": "large/logo",
                  "small": "small/logo",
                },
                "screenshots": Array [
                  Object {
                    "path": "screenshot",
                  },
                ],
                "updated": "2018-09-26",
                "version": "1",
              },
              "latestVersion": "1",
              "module": Object {},
              "name": "pretty cool plugin 1",
              "pinned": false,
              "state": "",
              "type": "",
            }
          }
          onModelChange={[Function]}
        />
        <div
          className="gf-form-group"
        />
        <ButtonRow
          isReadOnly={false}
          onDelete={[Function]}
          onSubmit={[Function]}
          onTest={[Function]}
        />
      </form>
    </div>
  </PageContents>
</Page>
`;

exports[`Render should render is ready only message 1`] = `
<Page
  navModel={Object {}}
>
  <PageContents
    isLoading={false}
  >
    <div>
      <form
        onSubmit={[Function]}
      >
        <div
          className="grafana-info-box span8"
        >
          This datasource was added by config and cannot be modified using the UI. Please contact your server admin to update this datasource.
        </div>
        <BasicSettings
          dataSourceName="gdev-cloudwatch"
          isDefault={false}
          onDefaultChange={[Function]}
          onNameChange={[Function]}
        />
        <PluginSettings
          dataSource={
            Object {
              "access": "",
              "basicAuth": false,
              "basicAuthPassword": "",
              "basicAuthUser": "",
              "database": "",
              "id": 13,
              "isDefault": false,
              "jsonData": Object {
                "authType": "credentials",
                "defaultRegion": "eu-west-2",
              },
              "name": "gdev-cloudwatch",
              "orgId": 1,
              "password": "",
              "readOnly": true,
              "type": "cloudwatch",
              "typeLogoUrl": "public/app/plugins/datasource/cloudwatch/img/amazon-web-services.png",
              "url": "",
              "user": "",
              "withCredentials": false,
            }
          }
          dataSourceMeta={
            Object {
              "defaultNavUrl": "some/url",
              "enabled": false,
              "hasUpdate": false,
              "id": "1",
              "info": Object {
                "author": Object {
                  "name": "Grafana Labs",
                  "url": "url/to/GrafanaLabs",
                },
                "description": "pretty decent plugin",
                "links": Array [
                  Object {
                    "name": "project",
                    "url": "one link",
                  },
                ],
                "logos": Object {
                  "large": "large/logo",
                  "small": "small/logo",
                },
                "screenshots": Array [
                  Object {
                    "path": "screenshot",
                  },
                ],
                "updated": "2018-09-26",
                "version": "1",
              },
              "latestVersion": "1",
              "module": Object {},
              "name": "pretty cool plugin 1",
              "pinned": false,
              "state": "",
              "type": "",
            }
          }
          onModelChange={[Function]}
        />
        <div
          className="gf-form-group"
        />
        <ButtonRow
          isReadOnly={true}
          onDelete={[Function]}
          onSubmit={[Function]}
          onTest={[Function]}
        />
      </form>
    </div>
  </PageContents>
</Page>
`;

exports[`Render should render loader 1`] = `
<Page
  navModel={Object {}}
>
  <PageContents
    isLoading={true}
  />
</Page>
`;
